src/Microsoft.ML.Data/DataView/CacheDataView.cs

```diff
@@ -1177,11 +1191,11 @@ protected sealed override void Dispose(bool disposing)
                 _disposed = true;
             }
 
-            public sealed override ValueGetter<TValue> GetGetter<TValue>(int col)
+            public sealed override ValueGetter<TValue> GetGetter<TValue>(int columnIndex)
             {
-                if (!IsColumnActive(col))
-                    throw Ch.Except("Column #{0} is requested but not active in the cursor", col);
-                var getter = _getters[_colToActivesIndex[col]] as ValueGetter<TValue>;
+                if (!IsColumnActive(columnIndex))
+                    throw Ch.Except("Column #{0} is requested but not active in the cursor", columnIndex);
```

Comment:
in how many different ways we can do same thing :)
`Ch.CheckParam(IsColumnActive(columnIndex), nameof(columnIndex), "requested column not active");`?

Reply:
in many. 

---
In reply to: [261485672](https://github.com/dotnet/machinelearning/pull/2796#discussion_r261485672) [](ancestors = 261485672)